backend/routers/lib/search.py

```diff
@@ -13,25 +12,32 @@
 MEILI_SERVER_URL = os.environ["MEILI_SERVER_URL"]
 MEILI_MASTER_KEY = os.environ["MEILI_MASTER_KEY"]
 
-search = meilisearch.Client(MEILI_SERVER_URL, MEILI_MASTER_KEY)
-if "companies" not in [index.uid for index in search.get_indexes()["results"]]:
-    search.create_index("companies", {"primaryKey": "cik"})
-    time.sleep(3)
-    search = meilisearch.Client(MEILI_SERVER_URL, MEILI_MASTER_KEY)
+
+def _prepare_meilisearch():
+    client = meilisearch_python_sdk.Client(MEILI_SERVER_URL, MEILI_MASTER_KEY)
+    companies_index = client.index("companies")
+    indexes = client.get_indexes()
+    if not indexes or "companies" not in [index.uid for index in indexes]:
+        task = client.create_index("companies", {"primaryKey": "cik"})
+        client.wait_for_task(task.task_uid, timeout=None)
+    companies_index.update_displayed_attributes(
+        [
+            "name",
+            "cik",
+            "tickers",
+        ]
+    )
+    companies_index.update_searchable_attributes(["name", "tickers", "cik"])
+    companies_index.update_filterable_attributes(["thirteen_f"])

```

Comment:
You are correct about the primary key and its fix...I thought I caught all these but I missed this one.

For the other issue, I don't know what I was thining. `meilisearch_python_sdk` automatically waits for the index creation unless you tell it not to so that isn't needed at all. As for the reason I went with `def` instead of `async def` here, FastAPI already has a running event loop so starting a second one could potentially lead to errors. Since this happens at startup I figured it wouldn't make any difference to go with sync code, avoiding the potential for an event loop error.

This update should fix the errors.

```suggestion
def _prepare_meilisearch():
    client = meilisearch_python_sdk.Client(MEILI_SERVER_URL, MEILI_MASTER_KEY)
    companies_index = client.index("companies")
    indexes = client.get_indexes()
    if not indexes or "companies" not in [index.uid for index in indexes]:
        client.create_index("companies", primary_key="cik")
    companies_index.update_displayed_attributes(
        [
            "name",
            "cik",
            "tickers",
        ]
    )
    companies_index.update_searchable_attributes(["name", "tickers", "cik"])
    companies_index.update_filterable_attributes(["thirteen_f"])
```